src/main/minio.js

```diff
@@ -1786,23 +1788,40 @@ export default class Client {
     return simpleUploader
   }
 
-   // Remove all the notification configurations in the S3 provider
-  setBucketNotification(bucketName, bucketNotification, cb) {
+  // Remove all the notification configurations in the S3 provider
+  setBucketNotification(bucketName, config, cb) {
     if (!isFunction(cb)) {
       throw new TypeError('callback should be of type "function"')
     }
-    var method = 'PUT'
-    var query = 'notification'
-    var builder = new xml2js.Builder({rootName:'NotificationConfiguration', renderOpts:{'pretty':false}, headless:true});
-    var payload = builder.buildObject(bucketNotification.data)
-    this.makeRequest({method, bucketName, query}, payload, 200, '', (e, response) => {
+    this.getBucketRegion(bucketName, (e, region) => {
       if (e) return cb(e)
-      cb(null)
+      let partition = 'minio'
+      if (isAmazonEndpoint(this.host)) partition = 'aws'
+      let updateARN = (target) => {
+	if (!target[target.arnTag]) {
+	  target[target.arnTag] = `arn:${partition}:${target.service}:${region}:${this.accessKey}:${target.resource}`
```

Comment:
I don't think this.accessKey is used in ARN, the user should supply the account id by himself, @harshavardhana ?